toolbox/check_certificates.py

```diff
@@ -0,0 +1,400 @@
+#!/opt/privacyidea/bin/python
+
+
+__doc__ = """
+Documentation of the Certificate Check Script
+
+Overview:
+
+This script is used to check TLS certificates for web servers and LDAP servers,
+as well as CA certificates. It allows you to check the validity of certificates
+and issue warnings if a certificate is about to expire.
+The script can optionally create log files.
+
+Usage:
+
+The script is executed from the command line
+and supports several arguments to configure the certificate checks.
+
+Command Line Arguments:
+
+- `--days`: (optional) Number of days before expiration to issue a warning.
+- `--days`: If not set, it will use 30 days as default.
+- `--config-dir`: Directory to search for web server configuration files.
+- `--web`: Checks the web server certificate.
+- `--ldap`: Checks the LDAP server certificate and corresponding CA certificate.
+- `--all`: Checks all web server, LDAP server, and CA certificates.
```

Comment:
Maybe we should make `--all` the default (check all configured certificates). Only if `--web` or `--ldap` is given, check this specific configuration.
Also: if `--ldap` is given, the certificates of all resolvers will be checked. We could make this optionally based on the resolver name. But we can do that later as well.